src/aggregation/metric/cardinality.rs

```diff
@@ -0,0 +1,381 @@
+        mut self,
+        agg_with_accessor: &AggregationWithAccessor,
+    ) -> crate::Result<IntermediateMetricResult> {
+        if self.column_type == ColumnType::Str {
+            let mut buffer = String::new();
+            let entries: Vec<u64> = self.entries.into_iter().collect();
+            let term_dict = agg_with_accessor.str_dict_column.as_ref().cloned().unwrap();
+            for term_id in entries {
+                if term_id == u64::MAX {
+                    let missing_key = self
+                        .missing
+                        .as_ref()
+                        .expect("Found placeholder term_id but `missing` is None");
+                    match missing_key {
+                        Key::Str(missing) => {
+                            self.cardinality.sketch.insert_any(&missing);
+                        }
+                        Key::F64(val) => {
+                            let val = f64_to_u64(*val);
+                            self.cardinality.sketch.insert_any(&val);
+                        }
+                    }
+                } else {
+                    if !term_dict.ord_to_str(term_id, &mut buffer)? {
+                        return Err(TantivyError::InternalError(format!(
+                            "Couldn't find term_id {term_id} in dict"
+                        )));
+                    }
+                    self.cardinality.sketch.insert_any(&buffer);
+                }
+            }
+        }
+
+        Ok(IntermediateMetricResult::Cardinality(self.cardinality))
+    }
+}
+
+impl SegmentAggregationCollector for SegmentCardinalityCollector {
+    fn add_intermediate_aggregation_result(
+        self: Box<Self>,
+        agg_with_accessor: &AggregationsWithAccessor,
+        results: &mut IntermediateAggregationResults,
+    ) -> crate::Result<()> {
+        let name = agg_with_accessor.aggs.keys[self.accessor_idx].to_string();
+        let agg_with_accessor = &agg_with_accessor.aggs.values[self.accessor_idx];
+
+        let intermediate_result = self.into_intermediate_metric_result(agg_with_accessor)?;
+        results.push(
+            name,
+            IntermediateAggregationResult::Metric(intermediate_result),
+        )?;
+
+        Ok(())
+    }
+
+    fn collect(
+        &mut self,
+        doc: crate::DocId,
+        agg_with_accessor: &mut AggregationsWithAccessor,
+    ) -> crate::Result<()> {
+        self.collect_block(&[doc], agg_with_accessor)
+    }
+
+    fn collect_block(
+        &mut self,
+        docs: &[crate::DocId],
+        agg_with_accessor: &mut AggregationsWithAccessor,
+    ) -> crate::Result<()> {
+        let bucket_agg_accessor = &mut agg_with_accessor.aggs.values[self.accessor_idx];
+        self.collect_block_with_field(docs, bucket_agg_accessor);
+
+        let col_block_accessor = &bucket_agg_accessor.column_block_accessor;
+        if self.column_type == ColumnType::Str {
+            for term_id in col_block_accessor.iter_vals() {
```

Comment:
```suggestion
            for term_ord in col_block_accessor.iter_vals() {
```

We call ordered ids ordinals in tantivy.
So here that would be `term_ord`.

Reply:
fixed